std/experimental/allocator/building_blocks/region.d

```diff
@@ -63,16 +63,16 @@ struct Region(ParentAllocator = NullAllocator,
         alias parent = ParentAllocator.instance;
     }
 
-    private void* _current, _begin, _end;
+    private BorrowedRegion!(minAlign, growDownwards) _impl;
 
     private void* roundedBegin() const pure nothrow @trusted @nogc
     {
-        return cast(void*) roundUpToAlignment(cast(size_t) _begin, alignment);
+        return _impl.roundedBegin;
```

Comment:
Instead of all of these forwardings can't we just use an `alias this` and simply delete them? That will simplify the code and also maybe make it a bit more efficient.

Reply:
1. Can't `alias this` a private member, and making `_impl` public would cause more problems (e.g., users could reassign it between allocations). Of course, there are other ways to forward methods, but...
2. Manual forwarding is the only way to get good docs. Even if we removed these forwardings from the actual implementation, we'd still need to keep the comments and method signatures around in a `version (StdDdoc)` block, so the savings would be marginal.